extensions/quartz/deployment/src/test/java/io/quarkus/quartz/test/DisabledSchedulerTest.java

```diff
@@ -0,0 +1,40 @@
+package io.quarkus.quartz.test;
+
+import static org.junit.jupiter.api.Assertions.assertFalse;
+
+import javax.inject.Inject;
+
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.asset.StringAsset;
+import org.jboss.shrinkwrap.api.spec.JavaArchive;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+import io.quarkus.quartz.runtime.QuartzScheduler;
+import io.quarkus.scheduler.Scheduled;
+import io.quarkus.test.QuarkusUnitTest;
+
+public class DisabledSchedulerTest {
+
+    @Inject
+    QuartzScheduler quartzScheduler;
```

Comment:
According to this https://github.com/quarkusio/quarkus/issues/8913#issuecomment-621144357, this should be `null` hence causing a NPE at https://github.com/quarkusio/quarkus/pull/9027/files#diff-b22e0da44e71f897e6899c8e8e9626fbR31 or am I missing something? 
/cc @mkouba

Reply:
+1

Reply:
So I believe that we should throw IllegalStateException from `io.quarkus.quartz.runtime.QuartzScheduler.produceQuartzScheduler()` if `QuartzScheduler.scheduler` is null. However, in this case we inject `io.quarkus.quartz.runtime.QuartzScheduler` which is OK. @adrianfiedler you should rather inject `io.quarkus.scheduler.Scheduler` which is part of the API (and is implemented by `QuartzScheduler`).

Reply:
Totally agree, how about "Cannot inject not started scheduler" as message for the ISE?"

Reply:
I would mention Quartz here. Something like "Cannot produce org.quartz.Scheduler - Quartz scheduler is disabled".

Reply:
Hm, looking at the CDI spec, we should already throw `IllegalProductException` because the producer is not `@Dependent` and returns `null`. Let me check this...

Reply:
Ah, we only throw `IllegalProductException` for normal scoped producers. Which is probably OK.

Reply:
There is also one additonal case where org.quartz.Scheduler is null: When no scheduled methods are found and `forceStart `config is `false`.

Reply:
Good point!

Reply:
So because the producer is @Singleton and it throws automatically IllegalProductException - no need to explicitly throw an ISE?

Reply:
Okay I double checked and injected `org.quartz.Scheduler` directly and did not get an `IllegalProductException`. I will throw then explicitly the ISE.

Reply:
Yes, `@Singleton` is not a normal scope. +1 for throwing ISE.